onset_maproom/tests/test_calc.py

```diff
@@ -3,11 +3,132 @@
 import xarray as xr
 
 import calc
+import data_test_calc
 
-def test_onset():
+
+def test_daily_tobegroupedby_season_cuts_on_days():
+
+    precip = data_test_calc.multi_year_data_sample()
+    dts = calc.daily_tobegroupedby_season(precip, 29, 11, 29, 2)
+
+    assert dts["T"].size == 461
+
+
+def test_daily_tobegroupedby_season_creates_groups():
+
+    precip = data_test_calc.multi_year_data_sample()
+    dts = calc.daily_tobegroupedby_season(precip, 29, 11, 29, 2)
+
+    assert dts["group"].size == 5
+
+
+def test_daily_tobegroupedby_season_picks_right_end_dates():
+
+    precip = data_test_calc.multi_year_data_sample()
+    dts = calc.daily_tobegroupedby_season(precip, 29, 11, 29, 2)
+    assert (
+        dts.seasons_ends
+        == pd.to_datetime(
+            xr.DataArray(
+                [
+                    "2001-02-28T00:00:00.000000000",
+                    "2002-02-28T00:00:00.000000000",
+                    "2003-02-28T00:00:00.000000000",
+                    "2004-02-29T00:00:00.000000000",
+                    "2005-02-28T00:00:00.000000000",
+                ],
+                dims=["group"],
+                coords={"group": dts.seasons_ends["group"]},
+            )
+        )
+    ).all()
+
+
+def test_seasonal_onset_date_keeps_returning_same_outputs():
+
+    precip = data_test_calc.multi_year_data_sample()
+    onsetsds = calc.seasonal_onset_date(
+        daily_rain=precip,
+        search_start_day=1,
+        search_start_month=3,
+        search_days=90,
+        wet_thresh=1,
+        wet_spell_length=3,
+        wet_spell_thresh=20,
+        min_wet_days=1,
+        dry_spell_length=7,
+        dry_spell_search=21,
+        time_coord="T",
+    )
+    onsets = onsetsds.onset_delta + onsetsds["T"]
+    assert np.isnat((onsets[0]))
+    assert onsets[1] == pd.to_datetime("2001-03-08T00:00:00.000000000")
+    assert np.isnat((onsets[2]))
+    assert onsets[3] == pd.to_datetime("2003-04-12T00:00:00.000000000")
+    assert onsets[4] == pd.to_datetime("2004-04-04T00:00:00.000000000")
+
+
+def test_seasonal_onset_date():
+    t = pd.date_range(start="2000-01-01", end="2005-02-28", freq="1D")
+    # this is rr_mrg.sel(T=slice("2000", "2005-02-28")).isel(X=150, Y=150).precip
+    synthetic_precip = xr.DataArray(np.zeros(t.size), dims=["T"], coords={"T": t}) + 1.1
```

Comment:
You can add `name="synthetic_precip"` here, rather than letting it use a default name and then renaming it later.

Reply:
xr.where resets the name to None...